### PR TITLE
refactor more properties to Sets

### PR DESCRIPTION
**Pull request by sokra**
**What kind of change does this PR introduce?**
refactoring

**Did you add tests for your changes?**
existing tests

**If relevant, link to documentation update:**
N/A

**Summary**
* Chunk.entrypoints -> Set
* Compilation.fileDependencies/contextDependencies/missingDependencies -> Set
* Params.compilationDependencies -> Set
* Module.fileDependencies/contextDependencies -> Set

**Does this PR introduce a breaking change?**
yes

**Other information**

### CODE REVIEW

File: lib/JsonpMainTemplatePlugin.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ class JsonpMainTemplatePlugin {
 
 	apply(mainTemplate) {
 		function needChunkLoadingCode(chunk) {
-			var otherChunksInEntry = chunk.entrypoints.some(function(entrypoint) {
+			var otherChunksInEntry = chunk.getEntrypoints().some(function(entrypoint) {
```
Comment (michael-ciniawsky) [inline lib/JsonpMainTemplatePlugin.js]:
```diff
- ...some(function(entrypoint) {
+ ...some(entrypoint => {
```

File: lib/Chunk.js [hunk 2]
```diff
@@ -597,4 +637,14 @@ Object.defineProperty(Chunk.prototype, "blocks", {
 	}, "Chunk.blocks: Use Chunk.addBlock/removeBlock/setBlocks to modify blocks.")
 });
 
+Object.defineProperty(Chunk.prototype, "entrypoints", {
```
Comment (TheLarkInn) [inline lib/Chunk.js]: Do we want to start using `Reflect.defineProperty` instead here or for any instance? I guess the only benefit is that it has friendlier returns (which in this case is pretty relevant.

Comment (TheLarkInn) [reply lib/Chunk.js]: Heh, actually check this out. https://twitter.com/bmeurer/status/928168764230848513

Comment (sokra) [reply lib/Chunk.js]: Throwing is expensive, but this line doesn't throw. It's not in the hot path anyway. Not worth changing.

File: lib/JsonpMainTemplatePlugin.js (hunk 1 above)
Comment (sokra) [reply lib/JsonpMainTemplatePlugin.js]: done in separate PR #5936

**Pull request merged**